src/main.rs

```diff
@@ -1,389 +1,129 @@
-    #[structopt(short = "o", long = "offset", default_value = "0")]
-    offset: f64,
-
-    /// How often to iterate. ex., 5s, 1h1m1s1ms1us
-    #[structopt(short = "e", long = "every", default_value = "1us",
-                parse(try_from_str = "parse_duration"))]
-    every: Duration,
-
-    /// A comma-separated list of values, placed into 4ITEM. ex., red,green,blue
-    #[structopt(long = "for", parse(from_str = "get_values"))]
-    ffor: Option<Vec<String>>,
-
-    /// Keep going until the duration has elapsed (example 1m30s)
-    #[structopt(short = "d", long = "for-duration", parse(try_from_str = "parse_duration"))]
-    for_duration: Option<Duration>,
-
-    /// Keep going until the output contains this string
-    #[structopt(short = "c", long = "until-contains")]
-    until_contains: Option<String>,
-
-    /// Keep going until the output changes
-    #[structopt(short = "C", long = "until-changes")]
-    until_changes: bool,
-
-    /// Keep going until the output changes
-    #[structopt(short = "S", long = "until-same")]
-    until_same: bool,
-
-    /// Keep going until the output matches this regular expression
-    #[structopt(short = "m", long = "until-match", parse(try_from_str = "Regex::new"))]
-    until_match: Option<Regex>,
-
-    /// Keep going until a future time, ex. "2018-04-20 04:20:00" (Times in UTC.)
-    #[structopt(short = "t", long = "until-time", parse(try_from_str = "parse_rfc3339_weak"))]
-    until_time: Option<SystemTime>,
-
-    /// Keep going until the command exit status is non-zero, or the value given
-    #[structopt(short = "r", long = "until-error", parse(from_str = "get_error_code"))]
-    until_error: Option<ErrorCode>,
-
-    /// Keep going until the command exit status is zero
-    #[structopt(short = "s", long = "until-success")]
-    until_success: bool,
-
-    /// Keep going until the command exit status is non-zero
-    #[structopt(short = "f", long = "until-fail")]
-    until_fail: bool,
-
-    /// Only print the output of the last execution of the command
-    #[structopt(short = "l", long = "only-last")]
-    only_last: bool,
-
-    /// Read from standard input
-    #[structopt(short = "i", long = "stdin")]
-    stdin: bool,
-
-    /// Exit with timeout error code on duration
-    #[structopt(short = "D", long = "error-duration")]
-    error_duration: bool,
 
-    /// Provide a summary
-    #[structopt(long = "summary")]
-    summary: bool,
-
-    /// The command to be looped
-    #[structopt(raw(multiple="true"))]
-    input: Vec<String>
-
-}
-
-fn precision_of(s: &str) -> usize {
-    let after_point = match s.find('.') {
-        // '.' is ASCII so has len 1
-        Some(point) => point + 1,
-        None => return 0,
-    };
-    let exp = match s.find(&['e', 'E'][..]) {
-        Some(exp) => exp,
-        None => s.len(),
-    };
-    exp - after_point
-}
-
-#[derive(Debug)]
-enum ErrorCode {
-    Any,
-    Code(u32),
+    process::exit(exit_status);
```

Comment:
the process::exit should stick in main so we can unit test other functions that we call into. Having it here will always exit the process.

Reply:
Yeah, that should be the case.